nestor_api/lib/workflow.py

```diff
@@ -49,3 +52,92 @@ def get_previous_step(project_config: dict, target: str) -> Optional[str]:
     if index > 0:
         return project_config["workflow"][index - 1]
     return None
+
+
+def init_workflow(
+    organization: str, app_name: str, git_provider: AbstractGitProvider
+) -> Tuple[str, Dict[str, Dict[str, Tuple[bool, bool]]]]:
+    """Initialize workflow for a given repo by creating all workflow branches.
+    This function is idempotent which means it will not try to recreate a
+    branch that already exist. However if a branch already exist but is not protected,
+    it will be set to protected."""
+
+    # Switch to staging environment in order to get application configuration
+    config.change_environment("staging")
+
+    # Get application configuration to get the list of workflow branches
+    app_config = config.get_app_config(app_name)
```

Comment:
If you choose to load the `app_config` from the route function, you could pass it as an argument and not have to retrieve it here.

Also, if you only need the workflow, maybe loading `project_config` is enough (it depends if we want to allow an app to override the default workflow, IMO we can allow it).

Reply:
I don't choose, this is a port of current implementation where it is working that way. Don't want to be involved in a massive refacto. Let's keep focus on the main objective which is to have something working quickly then improving it later ;)

Reply:
I don't see any difference to have it in argument rather than retrieving it here. This part is the lib responsible for the business logic whereas the route is just a dumb wrapper using it. Seems to me that it does not change anything.